tools/gluc/sources/main.cpp

```diff
@@ -1,36 +1,50 @@
 #include "Basic/SourceLocation.hpp"
 #include "Basic/SourceManager.hpp"
-#include "IRGen/IRGen.hpp"
 #include "Lexer/Scanner.hpp"
+#include "Parser/Parser.hpp"
 
-int main(int argc, char *argv[])
-{
-    if (argc != 2) {
-        llvm::errs() << "Usage: " << argv[0] << " <input file>\n";
-        return 1;
-    }
+#include "llvm/Support/CommandLine.h"
 
-    glu::SourceManager sourceManager;
+using namespace llvm;
+
+static cl::opt<bool> PrintAST(
+    "print-ast", cl::desc("Print the AST after parsing"), cl::init(false)
+);
 
-    auto fileID = sourceManager.loadFile(argv[1]);
+static cl::list<std::string> InputFilenames(
+    cl::Positional, cl::OneOrMore, cl::desc("<input glu files>")
+);
 
-    if (!fileID) {
-        llvm::errs() << fileID.getError().message() << "\n";
+int main(int argc, char **argv)
+{
+    cl::ParseCommandLineOptions(argc, argv);
+
+    if (InputFilenames.empty()) {
+        errs() << "Usage: " << argv[0] << " <input files...>\n";
         return 1;
     }
```

Comment:
this can't happen, InputFilenames is set as `OneOrMore` so the Parse above fails and exits first